### PR TITLE
boards: remove pulpino board

### PR DESCRIPTION
**Pull request by nashif**
This board is unmaintained and unsupported. It is not known to work and
has lots of conditional code across the tree that makes code
unmaintainable.

### CODE REVIEW

**Comment by andyross** [comment]
Yes, please.  This was the only legacy timer driver without a testable platform.  Once this merges I should go in and remove a bunch of the old glue code.

**Review by andyross — APPROVED** [review]

File: doc/release-notes-1.11.rst [hunk 1]
```diff
@@ -240,7 +240,7 @@ release:
 * :github:`3322` - [IPv6 TAHI] Section 3: RFC 4862 - IPv6 Stateless Address Autoconfiguration
 * :github:`3323` - [IPv6 TAHI] Section 2: RFC 4861 - Neighbor Discovery for IPv6
 * :github:`3329` - Build warnings [-Wpointer-sign] with LLVM/icx (bluetooth_ipsp)
-* :github:`3345` - Missing board documentation for riscv32/zedboard_pulpino
+* :github:`3345` - Missing board documentation for riscv32/
```
Comment (dbkinder) [inline doc/release-notes-1.11.rst]: Deleting release notes history doesn't seem right. The github issue 3345 remains, so should this release note.

**Review by dbkinder — DISMISSED** [review]

File: doc/release-notes-1.11.rst (hunk 1 above)
Comment (nashif) [reply doc/release-notes-1.11.rst]: addressed. was included by mistake.

**Comment by stefanct** [comment]
Since running Pulpino on Zedboards is a rather popular development platform in (my area of) academia the removal is a bit surprising to me. I am aware of the various problems maintaining support for legacy hardware brings but simply dropping support without much (public) discussion(?) does not foster any confidence TBH.

Is/was there any todo list indicating what would have been needed to keep it/update the worst offending parts? I can see some obvious problems/special cases for pulpino in the diff of this commit but some explanations to allow estimating the effort of fixing and re-adding support would be very nice - in case Zephyr is interested in that at all of course.

**Pull request merged**